src/EightBall.elm

```diff
@@ -565,11 +601,18 @@ scratch when =
 -}
 type WhatHappened
     = IllegalBreak (Pool AwaitingRack)
-    | PlayersFault (Pool AwaitingPlaceBallInHand)
+    | PlayersFault NextPlayerAction --SpotEightBall
+      -- | PlayersFaultSpotEight (Pool AwaitingPlaceBallInHand)
     | NextShot (Pool AwaitingPlayerShot)
     | GameOver (Pool AwaitingStart) { winner : Player }
 
 
+type NextPlayerAction
```

Comment:
This is the way I decided to handle different variations on `PlayersFault`.

Alternatively, we could have multiple top-level variants, like this:
```
      | PlayersFaultSpotEight (Pool AwaitingSpotEightBall)
```

The downside is that it would become awkward to add a `PlayersFaultChooseNextAction`.